[ENHANCEMENT] test on pandas <= 1.4.4

We want to test on pandas <= 1.4.4, but one unit test is currently breaking.

`phoenix.tests.metrics.test_wrappers.test_classification`

```
[precision_score-precision_score-parameters10-True-False-Y-df3] Failed: [undefined]TypeError: Cannot cast array data from dtype('O') to dtype('float64') according to the rule 'safe'
pos_label = 'Y'
df =          0x2a2e30ee 0x7901702 0x23a196fd 0x2e7394f9 0x2f3e4402
0.462154          Y  0.911863          Y   0.442004   0...        Y    0.83045       <NA>
0.968880          X  0.168375          Y        NaN   0.415101

[1000 rows x 5 columns]
metric = <ClassificationEval at 0x179310cc0 for function at 0x16b9bb370>
desired = <function precision_score at 0x16b9bb370>, parameters = {}
is_binary = True, pred_is_score = False

    @pytest.mark.parametrize(
        "pos_label,df",
        [
            ("Y", pd.DataFrame()),
            ("Y", gen_df(0)),
            ("Y", gen_df(N, 1)),
            ("Y", gen_df(N, NA_PCT, "XY")),
            ("Y", gen_df(N, NA_PCT, "XY", f=pd.Categorical)),
            ("Y", gen_df(N, NA_PCT, "XYZ")),
            ("Y", gen_df(N, NA_PCT, "XYZ", f=pd.Categorical)),
            (10, gen_df(N, NA_PCT, [9, 10])),
            (10, gen_df(N, NA_PCT, [9, 10], f=pd.Categorical)),
            (10, gen_df(N, NA_PCT, [9, 10, 11])),
            (10, gen_df(N, NA_PCT, [9, 10, 11], f=pd.Categorical)),
            (True, gen_df(N, NA_PCT, [False, True])),
            (True, gen_df(N, NA_PCT, [False, True], f=pd.Categorical)),
        ],
    )
    @pytest.mark.parametrize(
        "metric,desired,parameters,is_binary,pred_is_score",
        [
            (ev.accuracy_score, sk.accuracy_score, {}, False, False),
            (ev.accuracy_score, sk.accuracy_score, {}, True, False),
            (ev.average_precision_score, sk.average_precision_score, {}, True, True),
            (ev.balanced_accuracy_score, sk.balanced_accuracy_score, {}, True, False),
            (ev.brier_score_loss, sk.brier_score_loss, {}, True, True),
            (ev.f1_score, sk.f1_score, {}, True, False),
            (ev.hamming_loss, sk.hamming_loss, {}, True, False),
            (ev.jaccard_score, sk.jaccard_score, {}, True, False),
            (ev.log_loss, sk.log_loss, {}, True, True),
            (ev.matthews_corrcoef, sk.matthews_corrcoef, {}, True, False),
            (ev.precision_score, sk.precision_score, {}, True, False),
            (ev.recall_score, sk.recall_score, {}, True, False),
            (ev.roc_auc_score, sk.roc_auc_score, {}, True, True),
            (ev.zero_one_loss, sk.zero_one_loss, {}, True, False),
        ],
    )
    def test_classification(
        pos_label: Any,
        df: pd.DataFrame,
        metric: Eval,
        desired: Callable[..., float],
        parameters: Dict[str, Any],
        is_binary: bool,
        pred_is_score: bool,
    ) -> None:
>       run_test(
            df,
            actual_label_column_name,
            predicted_score_column_name if pred_is_score else predicted_label_column_name,
            sample_weight_column_name,
            metric,
            desired,
            parameters,
            actual_label=pos_label if is_binary else None,
            predicted_label=None if pred_is_score or not is_binary else pos_label,
        )

tests/metrics/test_wrappers.py:169: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/metrics/test_wrappers.py:246: in run_test
    desired_result = desired(
../miniforge3/envs/llmapps/lib/python3.10/site-packages/sklearn/metrics/_classification.py:1954: in precision_score
    p, _, _, _ = precision_recall_fscore_support(
../miniforge3/envs/llmapps/lib/python3.10/site-packages/sklearn/metrics/_classification.py:1577: in precision_recall_fscore_support
    MCM = multilabel_confusion_matrix(
../miniforge3/envs/llmapps/lib/python3.10/site-packages/sklearn/metrics/_classification.py:529: in multilabel_confusion_matrix
    tp_sum = np.bincount(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1... 1, 1,
       1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1,
       0, 1, 1, 0, 0, 1, 1, 1, 1, 1]),)
kwargs = {'minlength': 2, 'weights': array([0.590495419008875, 0.9876914107686465, 0.4833970643157919,
       0.481526549203327...5945032,
       0.6960230377719829, 0.14746456874597513, 0.9444387538987162,
       0.8308190977214496], dtype=object)}
relevant_args = (array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1...5945032,
       0.6960230377719829, 0.14746456874597513, 0.9444387538987162,
       0.8308190977214496], dtype=object))

>   ???
E   TypeError: Cannot cast array data from dtype('O') to dtype('float64') according to the rule 'safe'

<__array_function__ internals>:200: TypeError
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -97,6 +97,7 @@ artifacts = ["src/phoenix/server/static"]
 
 [tool.hatch.envs.default]
 dependencies = [
+  "pandas==1.4.0",
   "pytest",
   "pytest-cov",
   "pytest-lazy-fixture",

--- a/tests/metrics/test_wrappers.py
+++ b/tests/metrics/test_wrappers.py
@@ -30,7 +30,7 @@ def add_na(na_pct: float, series: "pd.Series[Any]") -> "pd.Series[Any]":
     series = series.sample(frac=1, random_state=rng)
     n_miss = int(len(series) * na_pct)
     series[: n_miss // 3] = None
-    series[n_miss // 3 : 2 * n_miss // 3] = pd.NA
+    series[n_miss // 3 : 2 * n_miss // 3] = float("nan")
     series[2 * n_miss // 3 : n_miss] = np.nan
     return series.sample(frac=1, ignore_index=True, random_state=rng)
 

--- a/tests/metrics/test_wrappers.py
+++ b/tests/metrics/test_wrappers.py
@@ -14,8 +14,8 @@ from phoenix.metrics.wrappers import Eval
 from phoenix.metrics.wrappers import SkEval as ev
 from sklearn import metrics as sk
 
-N = 1000
-NA_PCT = 0.1
+N = 32
+NA_PCT = 0.25
 
 rng = np.random.default_rng(1234567890)
 

